pkcs7/sign.go

```diff
@@ -11,40 +11,91 @@ import (
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/asn1"
+	"errors"
 	"io"
 	"time"
 )
 
-func Sign(r io.Reader, cert *x509.Certificate, priv *rsa.PrivateKey) ([]byte, error) {
-	hash := sha256.New()
-	if _, err := io.Copy(hash, r); err != nil {
+// Create Signature of message
+// message must be of type io.Reader or []byte
+// Returns signature and any error encountered.
+func Sign(message interface{}, certificate *x509.Certificate, privateKey *rsa.PrivateKey) ([]byte, error) {
```

Comment:
I don't think `message` should be changed to `interface{}`.

Instead, I'd suggest that Sign continues using `io.Reader` and the calling application can use `bytes.NewReader()` if it needs to pass in `[]bytes`.